internal/core/domain/unspent_repository.go

```diff
@@ -19,7 +19,7 @@ type UnspentRepository interface {
 	GetAvailableUnspents(ctx context.Context) ([]Unspent, error)
 	// GetAllUnspentsForAddresses returns the entire UTXO set (locked and spent
 	// included) for the provided list of addresses.
-	GetAllUnspentsForAddresses(ctx context.Context, addresses []string) ([]Unspent, error)
+	GetAllUnspentsForAddresses(ctx context.Context, addresses []string, page *Page) ([]Unspent, error)
```

Comment:
Based on name all objects should be retrieved and therefor Page object shouldnot be passed as arg.